src/library-authoring/LibraryLayout.tsx

```diff
@@ -13,23 +12,22 @@ import { ComponentEditorModal } from './components/ComponentEditorModal';
 const LibraryLayout = () => {
   const { libraryId } = useParams();
 
+  const match = useMatch('/library/:libraryId/collection/:collectionId');
+
+  const collectionId = match?.params.collectionId;
+
   if (libraryId === undefined) {
     // istanbul ignore next - This shouldn't be possible; it's just here to satisfy the type checker.
     throw new Error('Error: route is missing libraryId.');
   }
 
   return (
-    <LibraryProvider libraryId={libraryId}>
-      <Routes>
-        <Route
-          path="collection/:collectionId"
-          element={<LibraryCollectionPage />}
-        />
-        <Route
-          path="*"
-          element={<LibraryAuthoringPage />}
-        />
-      </Routes>
```

Comment:
We had to change to render based on a state instead of the route because of the ComponentPicker component

Reply:
Great! Fixed here: 1ce80cb5066939d307c8e68c9b815cf4deea262f